src/c_api.cpp

```diff
@@ -162,6 +163,7 @@ class Booster {
 
   void Predict(int num_iteration, int predict_type, int nrow,
                std::function<std::vector<std::pair<int, double>>(int row_idx)> get_row_fun,
+               PredictionEarlyStoppingHandle early_stop_handle,
```

Comment:
I think we cannot use function_pointer in c_api. It is hard to use this in python/R